secure-api-gateway-ob-uk-rs-server/src/main/java/com/forgerock/sapi/gateway/ob/uk/rs/server/api/obie/payment/v4_0_0/domesticstandingorders/DomesticStandingOrdersApiController.java

```diff
@@ -254,12 +254,13 @@ private OBWritePaymentDetailsResponse1 responseEntityDetails(FRDomesticStandingO
                         new OBWritePaymentDetailsResponse1Data()
                                 .addPaymentStatusItem(
                                         new OBWritePaymentDetails1()
-                                                .status(OBWritePaymentDetails1Status.valueOf(status.getValue()))
+                                                .status(status)
                                                 .paymentTransactionId(UUID.randomUUID().toString())
                                                 .statusUpdateDateTime(new DateTime(frStandingOrderSubmission.getUpdated()))
                                                 .statusDetail(
                                                         new OBWritePaymentDetails1StatusDetail()
-                                                                .status(OBWritePaymentDetails1StatusDetailStatus.valueOf(status.getValue()))
+                                                                .status(FRPaymentDetailsStatusConverter.toOBWritePaymentDetails1StatusDetailStatus(
+                                                                        frStandingOrderSubmission.getStatus().getValue()))
```

Comment:
Not possible this can be null?

Reply:
Already checked as status values come from an OB generated class (it's an enum).